### PR TITLE
Possible refactoring/switching to pathlib

### PR DESCRIPTION
**Pull request by rjboas**
This might only be a quick script for making directories but since you were interested this would be how I'd make some changes to polish it more.
Main things I've done are making separate functions, and switching to the more cross-platform and high-level [`pathlib`](https://docs.python.org/3/library/pathlib.html) library.
I also added a dry-run mode so you can check what it would create so you don't have to keep deleting folders if you are just testing file/folder names.
Some things I would also consider if I were making a more serious application or felt like adding would be command-line arguments (either quick way using `os` module or "properly" using [`argparse`](https://docs.python.org/3/howto/argparse.html#id1) and so on, and using a logging library instead of just `print` calls.

### CODE REVIEW

**Review by Tarang74 — APPROVED** [review]
Looks good

File: PythonScripts/UniFolderCreator/UniFolderCreator.py [hunk 1]
```diff
@@ -1,29 +1,48 @@
...
+
+def make_directories(parent: Path, weeks: int) -> None:
+    '''Creates a folder for each week in the specified parent.'''
+    for week in range(1, weeks+1):
+        new_folder_name = f"Week {week}"
+        full_path: Path = parent.joinpath(new_folder_name)
+
+        if full_path.exists():
+            print(f"WARNING:\tFolder {full_path} already exists!")
             pass
```
Comment (jboardman367) [inline PythonScripts/UniFolderCreator/UniFolderCreator.py:34]: Why the random pass here?

Comment (rjboas) [reply PythonScripts/UniFolderCreator/UniFolderCreator.py:34]: Whoops, yeah that doesn't need to be there. Must've left it in on accident

Comment (jboardman367) [reply PythonScripts/UniFolderCreator/UniFolderCreator.py:34]: also everything else looks great

**Comment by okyyy5** [comment]
Great tips and great improvements! I would like to integrate a proper logging framework and command line arguments to this, however, the scale of the script and general repository is far too small. Maybe in the future I will expand it with a GUI into an application with several small tools that people can utilize for everyday life.

**Pull request merged**